app/src/features/Sources/UploadSourceButton.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { ChangeEvent, useRef, useState } from "react";
+
+import { Button, Typography, makeStyles } from "@material-ui/core";
+import { useTranslation } from "react-i18next";
+
+import { useAppDispatch } from "app/store";
+import Alert from "common/components/Alert";
+import { useApiSourcesImportCreateMutation } from "services/api/endpoints";
+
+import { initSource } from "./sourceSlice";
+
+const useStyles = makeStyles((theme) => ({
+  fileInput: {
+    display: "none",
+  },
+  button: {
+    marginLeft: theme.spacing(3),
+    textTransform: "none",
+    width: "auto",
+    minWidth: 150,
+  },
+}));
+
+const UploadSourceButton = (): JSX.Element => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const dispatch = useAppDispatch();
+  const fileInputRef = useRef<HTMLInputElement | null>(null);
+  const [alert, setAlert] = useState<string | undefined>(undefined);
+  const [apiSourceImportCreate] = useApiSourcesImportCreateMutation();
+
+  let fileReader: FileReader | null = null;
+
+  const handleAlertClose = () => setAlert(undefined);
+  const handleFileRead = async () => {
+    const content = fileReader?.result;
+    if (content) {
+      const parsedContent = JSON.parse(content as string);
+      try {
+        await apiSourceImportCreate({ mappingRequest: parsedContent }).unwrap();
+        dispatch(initSource());
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (error: any) {
+        // TODO: Handle errors nicely
+        console.error(error);
+        if (error.data) {
```

Comment:
fyi `error` is of type `FetchBaseQueryError` (imported from `@reduxjs/toolkit/query`) and `data` of type `ApiValidationError` from `app/src/services/api/errors.ts`